config/confignet/confignet.go

```diff
@@ -36,20 +84,32 @@ type NetAddr struct {
 // Dial equivalent with net.Dialer's DialContext for this address.
 func (na *NetAddr) Dial(ctx context.Context) (net.Conn, error) {
 	d := net.Dialer{Timeout: na.DialerConfig.Timeout}
-	return d.DialContext(ctx, na.Transport, na.Endpoint)
+
+	tt := string(na.TransportType)
+	if na.Transport != "" {
+		tt = na.Transport
+	}
+
+	return d.DialContext(ctx, tt, na.Endpoint)
 }
 
 // Listen equivalent with net.ListenConfig's Listen for this address.
 func (na *NetAddr) Listen(ctx context.Context) (net.Listener, error) {
```

Comment:
Would also add a `Validate` for NetAddr and `NetTCP` since user may manually setup the transport (in code) and may be a wrong invalid value (typo).

Reply:
I'll add a `Validate` function for `NetAddr`, but `TCPAddr` doesn't allow users to set a `TransportType` so there isn't anything for us to Validate yet.